Add package org-tag-beautify

stardiviner opened the pull request:
### Brief summary of what the package does

This package use org-pretty-tags to prettify Org tags with icons or images.

### Direct link to the package repository

https://github.com/stardiviner/org-tag-beautify

### Your association with the package

maintainer

### Relevant communications with the upstream package maintainer

**No needed**

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Quick first pass --

- I wonder if it's an oversight that British Columbia is in the list of countries; also, should `uzbekistn`be written `uzbekistan`?
- `org-tag-beautify-enable` and `org-tag-beautify-disable` probably don't need to be autoloaded since you have an autoloaded minor mode defined, but I may be missing something
- org-tag-beautify.el#L12: Have you done the paperwork or is this copy-pasted?
- org-tag-beautify.el#L8: Prefer `;; Version` over `;; Package-Version` (MELPA automatically adds `Package-Version`)

stardiviner commented:
> Quick first pass --
> 
>     * I wonder if it's an oversight that British Columbia is in the list of countries; also, should `uzbekistn`be written `uzbekistan`?
> 
Not familiar with country names. I will check and update. Thanks for detailed checking.

>     * `org-tag-beautify-enable` and `org-tag-beautify-disable` probably don't need to be autoloaded since you have an autoloaded minor mode defined, but I may be missing something

You're right. I will remove those two autoload cookie.

> 
>     * org-tag-beautify.el#L12: Have you done the paperwork or is this copy-pasted?

Just copy-pasted, I will try to contact author of this work.

> 
>     * org-tag-beautify.el#L8: Prefer `;; Version` over `;; Package-Version` (MELPA automatically adds `Package-Version`)

I will update this.

Thanks 😄

stardiviner commented:
I fixed three improvements, Now only icons paper issue, I might try to find CC license icon images to replace this.

riscy commented:
> Just copy-pasted, I will try to contact author of this work.

The reason I bring that up is that sometimes people just copy-paste license/copyright boilerplate from a builtin package that FSF owns the copyright to.  If someone has gone to the trouble of doing the paperwork to assign their copyright, sometimes ELPA is the better choice for indexing the package.

If you can find CC images I think that would be best -- if you don't know where they came from, it's possible (however unlikely) we're not licensed to copy them around.

stardiviner commented:
> > Just copy-pasted, I will try to contact author of this work.
> 
> The reason I bring that up is that sometimes people just copy-paste license/copyright boilerplate from a builtin package that FSF owns the copyright to. If someone has gone to the trouble of doing the paperwork to assign their copyright, sometimes ELPA is the better choice for indexing the package.
> 
> If you can find CC images I think that would be best -- if you don't know where they came from, it's possible (however unlikely) we're not licensed to copy them around.

Sure, I'm finding solution of this license images issue. I will temporary remove those images. And only provide an option to trigger user to load custom data. Package will not relase shipped with licensed data.

Update code will be in two or three days.

riscy commented:
Thanks very much, and feel free to ping when ready.

stardiviner commented:
> org-tag-beautify.el#L12: Have you done the paperwork or is this copy-pasted?
Does this means have I assigned FSF papers? Yes, I assigned FSF papers.

stardiviner commented:
@riscy I removed the "data" directory in MELPA recipe now. And updated elisp code let user can specify the data directory. The problem should be solved.

riscy commented:
Great, thanks!
